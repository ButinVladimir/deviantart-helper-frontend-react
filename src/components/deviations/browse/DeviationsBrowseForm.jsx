import React from 'react';
import PropTypes from 'prop-types';
import DatePicker from 'react-datepicker';
import Section from 'react-bulma-components/lib/components/section';
import Container from 'react-bulma-components/lib/components/container';
import Content from 'react-bulma-components/lib/components/content';
import Heading from 'react-bulma-components/lib/components/heading';
import Columns from 'react-bulma-components/lib/components/columns';
import Button from 'react-bulma-components/lib/components/button';
import Pagination from 'react-bulma-components/lib/components/pagination';
import {
  Field,
  Label,
  Control,
  Input,
  Select,
  Checkbox,
} from 'react-bulma-components/lib/components/form';
import convertOptions from '../../../helpers/convert-options';
import { orderOptions, deviationsSortOptions } from '../../../consts/sort';
import { nsfwOptions } from '../../../consts/nsfw-options';
import { datePickerProps } from '../../../helpers/datepicker-props';

export default function DeviationsBrowseForm({
  page,
  pageCount,
  sortField,
  sortOrder,
  title,
  publishedTimeBegin,
  publishedTimeEnd,
  nsfw,
  filterByIds,
  pageLoading,
  showPagination,
  sortFieldChangeHandler,
  sortOrderChangeHandler,
  titleChangeHandler,
  publishedTimeBeginChangeHandler,
  publishedTimeEndChangeHandler,
  nsfwChangeHandler,
  filterByIdsChangeHandler,
  submitHandler,
  loadPageHandler,
}) {
  const publishedTimeBeginDate = publishedTimeBegin ? new Date(publishedTimeBegin) : null;
  const publishedTimeEndDate = publishedTimeEnd ? new Date(publishedTimeEnd) : null;
  const sortOptions = convertOptions(deviationsSortOptions);
  const orderOptionsElements = convertOptions(orderOptions);
  const nsfwOptionsElements = convertOptions(nsfwOptions);

  return (
    <Section>
      <Container>
        <Content>
          <Heading size={2}>Browse deviations</Heading>
        </Content>
        <Field>
          <Label>Title</Label>
          <Control>
            <Input
              id="title"
              disabled={pageLoading}
              name="title"
              value={title}
              onChange={titleChangeHandler}
            />
          </Control>
        </Field>

        <Columns>
          <Columns.Column>
            <Field>
              <Label>Published time range beginning</Label>
              <Control>
                <DatePicker
                  {...datePickerProps}
                  id="published-time-begin"
                  disabled={pageLoading}
                  selectsStart
                  name="publishedTimeBegin"
                  startDate={publishedTimeBeginDate}
                  endDate={publishedTimeEndDate}
                  selected={publishedTimeBeginDate}
                  onChange={publishedTimeBeginChangeHandler}
                />
              </Control>
            </Field>
          </Columns.Column>

          <Columns.Column>
            <Field>
              <Label>Published time range end</Label>
              <Control>
                <DatePicker
                  {...datePickerProps}
                  id="published-time-end"
                  disabled={pageLoading}
                  selectsEnd
                  name="publishedTimeEnd"
                  startDate={publishedTimeBeginDate}
                  endDate={publishedTimeEndDate}
                  selected={publishedTimeEndDate}
                  onChange={publishedTimeEndChangeHandler}
                />
              </Control>
            </Field>
          </Columns.Column>

          <Columns.Column>
            <Field>
              <Label>Sort by</Label>
              <Control>
                <Select
                  id="sort-field"
                  disabled={pageLoading}
                  name="sortField"
                  value={sortField}
                  onChange={sortFieldChangeHandler}
                >
                  {sortOptions}
                </Select>
              </Control>
            </Field>
          </Columns.Column>

          <Columns.Column narrow>
            <Field>
              <Label>Sort order</Label>
              <Control>
                <Select
                  id="sort-order"
                  disabled={pageLoading}
                  name="sortOrder"
                  value={sortOrder}
                  onChange={sortOrderChangeHandler}
                >
                  {orderOptionsElements}
                </Select>
              </Control>
            </Field>
          </Columns.Column>
        </Columns>

        <Columns>
          <Columns.Column>
            <Field>
              <Label>NSFW</Label>
              <Control>
                <Select
                  id="nsfw"
                  disabled={pageLoading}
                  name="nsfw"
                  value={nsfw}
                  onChange={nsfwChangeHandler}
                >
                  {nsfwOptionsElements}
                </Select>
              </Control>
            </Field>
          </Columns.Column>

          <Columns.Column narrow>
            <Field>
              <Label>Filter by selected</Label>
              <Control>
                <Checkbox
                  id="filter-by-ids"
                  disabled={pageLoading}
                  name="filterbyselected"
                  checked={filterByIds}
                  onChange={filterByIdsChangeHandler}
                />
              </Control>
            </Field>
          </Columns.Column>
        </Columns>

        <Columns>
          <Columns.Column>
            {showPagination && (
              <Pagination
                id="pagination"
                // Pagination in Bulma starts from 1 while pagination on backend start from 0.
                current={page + 1}
                total={pageCount}
                delta={3}
                onChange={loadPageHandler}
              />
            )}
          </Columns.Column>
          <Columns.Column narrow>
            <Field kind="group" align="right">
              <Control>
                <Button id="submit-button" color="primary" onClick={submitHandler} loading={pageLoading}>
                  {showPagination ? 'Refresh' : 'Submit' }
                </Button>
              </Control>
            </Field>
          </Columns.Column>
        </Columns>

      </Container>
    </Section>
  );
}

DeviationsBrowseForm.propTypes = {
  page: PropTypes.number.isRequired,
  pageCount: PropTypes.number.isRequired,
  sortField: PropTypes.string.isRequired,
  sortOrder: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  publishedTimeBegin: PropTypes.number,
  publishedTimeEnd: PropTypes.number,
  nsfw: PropTypes.string.isRequired,
  filterByIds: PropTypes.bool.isRequired,
  pageLoading: PropTypes.bool.isRequired,
  showPagination: PropTypes.bool.isRequired,
  sortFieldChangeHandler: PropTypes.func.isRequired,
  sortOrderChangeHandler: PropTypes.func.isRequired,
  titleChangeHandler: PropTypes.func.isRequired,
  publishedTimeBeginChangeHandler: PropTypes.func.isRequired,
  publishedTimeEndChangeHandler: PropTypes.func.isRequired,
  nsfwChangeHandler: PropTypes.func.isRequired,
  filterByIdsChangeHandler: PropTypes.func.isRequired,
  submitHandler: PropTypes.func.isRequired,
  loadPageHandler: PropTypes.func.isRequired,
};

DeviationsBrowseForm.defaultProps = {
  publishedTimeBegin: null,
  publishedTimeEnd: null,
};
