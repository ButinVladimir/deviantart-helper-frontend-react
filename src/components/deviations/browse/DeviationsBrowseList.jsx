import React from 'react';
import PropTypes from 'prop-types';
import Section from 'react-bulma-components/lib/components/section';
import Container from 'react-bulma-components/lib/components/container';
import DeviationsPreview from '../shared/DeviationsPreview';

export default function DeviationsBrowseList({
  deviations,
  selectedIds,
  toggleSelectionHandler,
}) {
  const mappedDeviations = deviations.map(d => (
    <DeviationsPreview
      key={d.id}
      selected={selectedIds.includes(d.id)}
      toggleSelectionHandler={toggleSelectionHandler}
      {...d}
    />
  ));

  return (
    <Section>
      <Container>
        {mappedDeviations}
      </Container>
    </Section>
  );
}


DeviationsBrowseList.propTypes = {
  deviations: PropTypes.arrayOf(PropTypes.any).isRequired,
  selectedIds: PropTypes.arrayOf(PropTypes.string).isRequired,
  toggleSelectionHandler: PropTypes.func.isRequired,
};
