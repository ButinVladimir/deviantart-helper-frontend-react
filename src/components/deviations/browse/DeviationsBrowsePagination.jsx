import React from 'react';
import PropTypes from 'prop-types';
import Section from 'react-bulma-components/lib/components/section';
import Container from 'react-bulma-components/lib/components/container';
import Pagination from 'react-bulma-components/lib/components/pagination';

export default function DeviationsBrowsePagination({
  page,
  pageCount,
  showPagination,
  loadPageHandler,
}) {
  return (
    <Section>
      <Container>
        {showPagination && (
          <Pagination
            // Pagination in Bulma starts from 1 while pagination on backend start from 0.
            current={page + 1}
            total={pageCount}
            delta={3}
            onChange={loadPageHandler}
          />
        )}
      </Container>
    </Section>
  );
}

DeviationsBrowsePagination.propTypes = {
  page: PropTypes.number.isRequired,
  pageCount: PropTypes.number.isRequired,
  showPagination: PropTypes.bool.isRequired,
  loadPageHandler: PropTypes.func.isRequired,
};
