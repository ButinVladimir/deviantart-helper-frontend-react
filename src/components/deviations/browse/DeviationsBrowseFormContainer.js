import { connect } from 'react-redux';
import deviationsBrowseChangeTitleActionCreator from '../../../redux/action-creators/deviations/browse/change-title';
import deviationsBrowseChangePublishedTimeBeginActionCreator from '../../../redux/action-creators/deviations/browse/change-published-time-begin';
import deviationsBrowseChangePublishedTimeEndActionCreator from '../../../redux/action-creators/deviations/browse/change-published-time-end';
import deviationsBrowseChangeSortFieldActionCreator from '../../../redux/action-creators/deviations/browse/change-sort-field';
import deviationsBrowseChangeSortOrderActionCreator from '../../../redux/action-creators/deviations/browse/change-sort-order';
import deviationsBrowseLoadPageActionCreator, { loadPage } from '../../../redux/action-creators/deviations/browse/load-page';
import DeviationsBrowseForm from './DeviationsBrowseForm';

/**
 * @description
 * Maps Redux state to props.
 *
 * @param {State} state - The state.
 * @returns {Object} Props.
 */
const mapStateToProps = state => ({
  page: state.deviations.browse.page,
  pageCount: state.deviations.browse.pageCount,
  sortField: state.deviations.browse.sortField,
  sortOrder: state.deviations.browse.sortOrder,
  title: state.deviations.browse.title,
  publishedTimeBegin: state.deviations.browse.publishedTimeBegin,
  publishedTimeEnd: state.deviations.browse.publishedTimeEnd,
  pageLoading: state.deviations.browse.pageLoading,
  showPagination: state.deviations.browse.showPagination,
});

/**
 * @description
 * Maps Redux dispatch to props.
 *
 * @param {Function} dispatch - The dispatch.
 * @param {Object} ownProps - The own component props.
 * @returns {Object} Props.
 */
const mapDispatchToProps = (dispatch, ownProps) => ({
  titleChangeHandler: e => dispatch(deviationsBrowseChangeTitleActionCreator(e.target.value)),
  publishedTimeBeginChangeHandler:
    value => dispatch(deviationsBrowseChangePublishedTimeBeginActionCreator(value)),
  publishedTimeEndChangeHandler:
    value => dispatch(deviationsBrowseChangePublishedTimeEndActionCreator(value)),
  sortFieldChangeHandler:
    e => dispatch(deviationsBrowseChangeSortFieldActionCreator(e.target.value)),
  sortOrderChangeHandler:
    e => dispatch(deviationsBrowseChangeSortOrderActionCreator(e.target.value)),
  submitHandler: () => dispatch(deviationsBrowseLoadPageActionCreator(ownProps.config)),
  // Pagination in Bulma starts from 1 while pagination on backend start from 0.
  loadPageHandler: page => dispatch(loadPage(page - 1, ownProps.config)),
});

export default connect(mapStateToProps, mapDispatchToProps)(DeviationsBrowseForm);
