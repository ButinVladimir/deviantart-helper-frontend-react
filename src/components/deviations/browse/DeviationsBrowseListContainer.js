import { connect } from 'react-redux';
import DeviationsBrowseList from './DeviationsBrowseList';
import deviationsBrowseLoadPageActionCreator from '../../../redux/action-creators/deviations/browse/load-page';

/**
 * @description
 * Maps Redux state to props.
 *
 * @param {State} state - The state.
 * @returns {Object} Props.
 */
const mapStateToProps = state => ({
  deviations: state.deviations.browse.deviations,
});

/**
 * @description
 * Maps Redux dispatch to props.
 *
 * @param {Function} dispatch - The dispatch.
 * @param {Object} ownProps - The own component props.
 * @returns {Object} Props.
 */
const mapDispatchToProps = (dispatch, ownProps) => ({
  preloadDeviationsHandler: () => dispatch(deviationsBrowseLoadPageActionCreator(ownProps.config)),
});

export default connect(mapStateToProps, mapDispatchToProps)(DeviationsBrowseList);
