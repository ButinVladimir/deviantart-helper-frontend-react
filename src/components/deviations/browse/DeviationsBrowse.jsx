import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import DeviationsBrowseForm from './DeviationsBrowseFormContainer';
import DeviationsBrowseList from './DeviationsBrowseListContainer';
import DeviationsBrowsePagination from './DeviationsBrowsePaginationContainer';
import Config from '../../../config/config';

export default class DeviationsBrowse extends Component {
  componentDidMount() {
    const { clearDataHander } = this.props;
    clearDataHander();
  }

  render() {
    const { config } = this.props;

    return (
      <Fragment>
        <DeviationsBrowseForm config={config} />
        <DeviationsBrowseList config={config} />
        <DeviationsBrowsePagination config={config} />
      </Fragment>
    );
  }
}

DeviationsBrowse.propTypes = {
  config: PropTypes.instanceOf(Config).isRequired,
  clearDataHander: PropTypes.func.isRequired,
};
