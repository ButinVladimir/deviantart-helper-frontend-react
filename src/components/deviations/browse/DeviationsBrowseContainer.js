import { connect } from 'react-redux';
import clearLoadedDataActionCreator from '../../../redux/action-creators/shared/clear-loaded-data';
import DeviationsBrowse from './DeviationsBrowse';

/**
 * @description
 * Maps Redux state to props.
 *
 * @returns {Object} Props.
 */
const mapStateToProps = () => ({
});

/**
 * @description
 * Maps Redux dispatch to props.
 *
 * @param {Function} dispatch - The dispatch.
 * @returns {Object} Props.
 */
const mapDispatchToProps = dispatch => ({
  clearDataHander: () => dispatch(clearLoadedDataActionCreator()),
});

export default connect(mapStateToProps, mapDispatchToProps)(DeviationsBrowse);
