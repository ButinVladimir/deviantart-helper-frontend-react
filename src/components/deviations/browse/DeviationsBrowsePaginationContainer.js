import { connect } from 'react-redux';
import { loadPage } from '../../../redux/action-creators/deviations/browse/load-page';
import DeviationsBrowsePagination from './DeviationsBrowsePagination';

/**
 * @description
 * Maps Redux state to props.
 *
 * @param {State} state - The state.
 * @returns {Object} Props.
 */
const mapStateToProps = state => ({
  page: state.deviations.browse.page,
  pageCount: state.deviations.browse.pageCount,
  showPagination: state.deviations.browse.showPagination,
});

/**
 * @description
 * Maps Redux dispatch to props.
 *
 * @param {Function} dispatch - The dispatch.
 * @param {Object} ownProps - The own component props.
 * @returns {Object} Props.
 */
const mapDispatchToProps = (dispatch, ownProps) => ({
  // Pagination in Bulma starts from 1 while pagination on backend start from 0.
  loadPageHandler: page => dispatch(loadPage(page - 1, ownProps.config)),
});

export default connect(mapStateToProps, mapDispatchToProps)(DeviationsBrowsePagination);
