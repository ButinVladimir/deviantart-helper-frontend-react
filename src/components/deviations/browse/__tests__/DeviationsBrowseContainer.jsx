import React from 'react';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import config from '../../../../config';
import DeviationsBrowseContainer from '../DeviationsBrowseContainer';
import { getConfigProvider } from '../../../shared/ConfigContext';
import createDefaultState from '../../../../redux/states/state';
import clearLoadedDataActionCreator from '../../../../redux/action-creators/shared/clear-loaded-data';
import { deviationsBrowseLoadCurrentPageActionCreator } from '../../../../redux/action-creators/deviations/browse/load-page';

jest.mock(
  '../../../../redux/action-creators/shared/clear-loaded-data',
  () => jest.fn(() => ({ type: 'CLEAR_LOADED_DATA' })),
);
jest.mock(
  '../../../../redux/action-creators/deviations/browse/load-page',
  () => ({
    __esModule: true,
    deviationsBrowseLoadCurrentPageActionCreator: jest.fn(() => ({ type: 'LOAD_PAGE' })),
  }),
);

describe('DeviationsBrowseContainer', () => {
  it('can be rendered correctly', () => {
    const state = createDefaultState();

    const mockStore = configureStore([thunk]);
    const store = mockStore(state);

    const ConfigProvider = getConfigProvider();
    const wrapper = mount(
      <ConfigProvider value={config}>
        <Provider store={store}>
          <DeviationsBrowseContainer />
        </Provider>
      </ConfigProvider>,
    );

    const actions = store.getActions();

    expect(wrapper).toMatchSnapshot();
    expect(clearLoadedDataActionCreator).toHaveBeenCalled();
    expect(deviationsBrowseLoadCurrentPageActionCreator).toHaveBeenCalled();
    expect(actions).toEqual([
      { type: 'CLEAR_LOADED_DATA' },
      { type: 'LOAD_PAGE' },
    ]);
  });
});
