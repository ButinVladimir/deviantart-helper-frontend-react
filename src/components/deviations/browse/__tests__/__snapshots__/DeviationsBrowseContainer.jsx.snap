// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviationsBrowseContainer can be rendered correctly 1`] = `
Array [
  <section
    className="section"
    style={Object {}}
  >
    <div
      className="container"
      style={Object {}}
    >
      <div
        className="content"
        style={Object {}}
      >
        <h1
          className="title is-2"
        >
          Browse deviations
        </h1>
      </div>
      <div
        className="field"
        style={Object {}}
      >
        <label
          className="label"
          htmlFor={null}
          style={Object {}}
        >
          Title
        </label>
        <div
          className="control"
          style={Object {}}
        >
          <input
            className="input"
            disabled={false}
            id="title"
            name="title"
            onChange={[Function]}
            placeholder=""
            readOnly={false}
            style={Object {}}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="columns is-multiline"
        style={Object {}}
      >
        <div
          className="column"
          style={Object {}}
        >
          <div
            className="field"
            style={Object {}}
          >
            <label
              className="label"
              htmlFor={null}
              style={Object {}}
            >
              Published time range beginning
            </label>
            <div
              className="control"
              style={Object {}}
            >
              <div>
                <div
                  className="react-datepicker__input-container"
                >
                  <input
                    autoComplete="off"
                    className="input"
                    disabled={false}
                    id="published-time-begin"
                    name="publishedTimeBegin"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    readOnly={false}
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="column"
          style={Object {}}
        >
          <div
            className="field"
            style={Object {}}
          >
            <label
              className="label"
              htmlFor={null}
              style={Object {}}
            >
              Published time range end
            </label>
            <div
              className="control"
              style={Object {}}
            >
              <div>
                <div
                  className="react-datepicker__input-container"
                >
                  <input
                    autoComplete="off"
                    className="input"
                    disabled={false}
                    id="published-time-end"
                    name="publishedTimeEnd"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    readOnly={false}
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="column"
          style={Object {}}
        >
          <div
            className="field"
            style={Object {}}
          >
            <label
              className="label"
              htmlFor={null}
              style={Object {}}
            >
              Sort by
            </label>
            <div
              className="control"
              style={Object {}}
            >
              <div
                className="select"
                style={Object {}}
              >
                <select
                  disabled={false}
                  id="sort-field"
                  multiple={false}
                  name="sortField"
                  onChange={[Function]}
                  readOnly={false}
                  value="publishedTime"
                >
                  <option
                    key="title"
                    value="title"
                  >
                    Title
                  </option>
                  <option
                    key="publishedTime"
                    value="publishedTime"
                  >
                    Published time
                  </option>
                  <option
                    key="views"
                    value="views"
                  >
                    Views
                  </option>
                  <option
                    key="favourites"
                    value="favourites"
                  >
                    Favourites
                  </option>
                  <option
                    key="comments"
                    value="comments"
                  >
                    Comments
                  </option>
                  <option
                    key="downloads"
                    value="downloads"
                  >
                    Downloads
                  </option>
                </select>
              </div>
            </div>
          </div>
        </div>
        <div
          className="column is-narrow"
          style={Object {}}
        >
          <div
            className="field"
            style={Object {}}
          >
            <label
              className="label"
              htmlFor={null}
              style={Object {}}
            >
              Sort order
            </label>
            <div
              className="control"
              style={Object {}}
            >
              <div
                className="select"
                style={Object {}}
              >
                <select
                  disabled={false}
                  id="sort-order"
                  multiple={false}
                  name="sortOrder"
                  onChange={[Function]}
                  readOnly={false}
                  value={-1}
                >
                  <option
                    key="1"
                    value={1}
                  >
                    Ascending
                  </option>
                  <option
                    key="-1"
                    value={-1}
                  >
                    Descending
                  </option>
                </select>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="columns is-multiline"
        style={Object {}}
      >
        <div
          className="column"
          style={Object {}}
        >
          <div
            className="field"
            style={Object {}}
          >
            <label
              className="label"
              htmlFor={null}
              style={Object {}}
            >
              NSFW
            </label>
            <div
              className="control"
              style={Object {}}
            >
              <div
                className="select"
                style={Object {}}
              >
                <select
                  disabled={false}
                  id="nsfw"
                  multiple={false}
                  name="nsfw"
                  onChange={[Function]}
                  readOnly={false}
                  value="SHOW_ALL"
                >
                  <option
                    key="SHOW_ALL"
                    value="SHOW_ALL"
                  >
                    Show everything
                  </option>
                  <option
                    key="SHOW_NON_NSFW"
                    value="SHOW_NON_NSFW"
                  >
                    Show only non-NSFW
                  </option>
                  <option
                    key="SHOW_NSFW"
                    value="SHOW_NSFW"
                  >
                    Show only NSFW
                  </option>
                </select>
              </div>
            </div>
          </div>
        </div>
        <div
          className="column is-narrow"
          style={Object {}}
        >
          <div
            className="field"
            style={Object {}}
          >
            <label
              className="label"
              htmlFor={null}
              style={Object {}}
            >
              Filter by selected
            </label>
            <div
              className="control"
              style={Object {}}
            >
              <label
                className="checkbox"
                disabled={false}
                style={Object {}}
              >
                <input
                  checked={false}
                  disabled={false}
                  id="filter-by-ids"
                  name="filterbyselected"
                  onChange={[Function]}
                  type="checkbox"
                  value=""
                />
              </label>
            </div>
          </div>
        </div>
      </div>
      <div
        className="columns is-multiline"
        style={Object {}}
      >
        <div
          className="column"
          style={Object {}}
        />
        <div
          className="column is-narrow"
          style={Object {}}
        >
          <div
            className="field is-grouped is-grouped-right"
            style={Object {}}
          >
            <div
              className="control"
              style={Object {}}
            >
              <button
                className="is-primary button"
                disabled={false}
                id="submit-button"
                onClick={[Function]}
                style={Object {}}
                tabIndex={0}
              >
                Submit
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <section
    className="section"
    style={Object {}}
  >
    <div
      className="container"
      style={Object {}}
    />
  </section>,
  <section
    className="section"
    style={Object {}}
  >
    <div
      className="container"
      style={Object {}}
    />
  </section>,
]
`;
